### PR TITLE
document #[panic_handler]

### PR DESCRIPTION
**Pull request by japaric**
r? @steveklabnik 

Required by rust-lang/rust#51366

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
@steveklabnik Can we get a review here today? I'd like to merge this quickly so we can have docs ready to go for Edition RC.

**Comment by japaric** [comment]
This needs to be updated because it's still referring to `panic_implementation`, the old name of `panic_handler`. But I gotta run right now so I'll get back to this in a few hours.

**Comment by steveklabnik** [comment]
Ping me when that’s done! I’ll make sure it gets in.

**Comment by japaric** [comment]
@steveklabnik updated!

File: src/panic-handler.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+## #[panic_handler]
+
+`#[panic_handler]` is used to define the behavior of `panic!` in `#![no_std]` applications.
+The `#[panic_handler]` attribute must be applied to a function with signature `fn(&PanicInfo)
+-> !` and such function must appear *once* in the dependency graph of a binary / dylib / cdylib
+crate. The API of `PanicInfo` can be found in the [API docs].
+
+[API docs]: https://doc.rust-lang.org/nightly/core/panic/struct.PanicInfo.html
```
Comment (steveklabnik) [inline src/panic-handler.md]: This should be a relative link so it works offline

    ../core/panic/struct.PanicInfo.html

Comment (japaric) [reply src/panic-handler.md]: Fixed!

**Comment by japaric** [comment]
Err, I forgot that mdbook doesn't support no_std code so I have just marked the code snippets as ignored.

**Comment by steveklabnik** [comment]
Awesome, thanks!

**Pull request merged**